### PR TITLE
Add Rails 7.1 support

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

This PR makes the changes necessary to support Rails 7.1.

### Integration

No changes necessary in production.

#### List the issues that this change affects.

Fixes https://github.com/primer/view_components/issues/2318

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?

The only thing necessary to support Rails 7.1 is to remove an unnecessary `require` statement in the Rails engine. Rails 7.1 no longer adds eager load paths to Ruby's load path, preferring instead to let Zeitwerk autoload missing constants.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### CODE REVIEW

**Comment by safeforge** [comment]
@camertron Removing the `require "primer/form_helper"` does not appear to be sufficient, but you have correctly identified the issue.

I'm experiencing issues with `Form` usage, and the error appears to be similar to the previous one: "cannot load such file -- primer/class_name_helper."  

Indeed, by removing `require "primer/class_name_helper"` from the file `lib/primer/forms/base_component.rb`, the issue with my application appears to be resolved. I wonder if there are any other `require` causing errors.

**Comment by camertron** [comment]
@safeforge interesting, thanks for the heads-up. Let me see if I can get our test suite to fail, that would really help catch these issues.

**Comment by camertron** [comment]
Ok, I totally forgot to bump the framework defaults in the demo app's application.rb. Doing so uncovered several additional bad `require` statements that I have since removed.

@safeforge can you verify this addresses the issues you're seeing?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,6 +51,8 @@ jobs:
             ruby_version: "2.7"
```
Comment (jonrohan) [inline .github/workflows/test.yml:51]: How do you feel about also dropping this matrix?

Comment (camertron) [reply .github/workflows/test.yml:51]: I'm ok with that 👍

Comment (jonrohan) [reply .github/workflows/test.yml:51]: I like the simplicity of having 3 groups, and adding a new one makes me want to drop the old one. I don’t know what sorta social contract we have to the community to always test against 6.1. I say let's go for it cause it lets us move our dependencies forward

Comment (camertron) [reply .github/workflows/test.yml:51]: Decision: leave 6.1 support in as long as it's easy. Remove it as soon as it's inconvenient.

**Review by jonrohan — APPROVED** [review]
👍🏻

**Comment by safeforge** [comment]
> @safeforge can you verify this addresses the issues you're seeing?

@camertron, I can confirm that your fix has resolved the issue in my use case. Thank you!

**Pull request merged**